platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -141,7 +145,7 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
             )
         } else {
             val configurationCache = isolatedProjects || startParameter.configurationCache.get()
-            val configureOnDemand = !isolatedProjects && startParameter.isConfigureOnDemand
+            val configureOnDemand = (!isolatedProjects || isolatedProjects && options.getOption(isolatedProjectsTasksConfigureOnDemand).get()) && startParameter.isConfigureOnDemand
```

Comment:
Or do we say that `isolatedProjectsTasksConfigureOnDemand ` is enough, no `startParameter.isConfigureOnDemand` respect is needed?

Reply:
Enabling IP should eventually imply incrementality, which implies `val configureOnDemand` should be true. In this PR we are saying that for the IP to imply incrementality (when running tasks) an extra internal flag should be enabled.

For all the brevity of the complex logical expressions, they often become cryptic, hiding the original intent.
My take on the condition would be:
```
val configureOnDemand = if (isolatedProjects) new_internal_flag else startParameter.configureOnDemand
```

Reply:
The `if` formulation is clear 👍